hs_tracking/models.py

```diff
@@ -0,0 +1,147 @@
+                    except Visitor.DoesNotExist:
+                        session.visitor.user = request.user
+                        session.visitor.save()
+                return session
+        # No session found, create one
+        if request.user.is_authenticated():
+            visitor, _ = Visitor.objects.get_or_create(user=request.user)
+        else:
+            visitor = Visitor.objects.create()
+        session = Session.objects.create(visitor=visitor)
+        request.session['hs_tracking_id'] = signing.dumps({'id': session.id})
+        return session
+
+
+class Visitor(models.Model):
+    first_seen = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, null=True, blank=True, unique=True)
+
+    def export_visitor_information(self):
+        """Exports visitor profile information."""
+        info = {
+            "id": self.id,
+        }
+        if self.user:
+            profile = UserProfile.objects.get(user=self.user)
+            for field in PROFILE_FIELDS:
+                info[field] = getattr(profile, field)
+            for field in USER_FIELDS:
+                info[field] = getattr(self.user, field)
+        else:
+            profile = None
+            for field in PROFILE_FIELDS:
+                info[field] = None
+            for field in USER_FIELDS:
+                info[field] = None
+        return info
+
+
+class Session(models.Model):
+    begin = models.DateTimeField(auto_now_add=True)
+    visitor = models.ForeignKey(Visitor)
+
+    objects = SessionManager()
+
+    def get(self, name):
+        return Variable.objects.filter(session=self, name=name).first().get_value()
+
+    def getlist(self, name):
+        return [v.get_value() for v in Variable.objects.filter(session=self, name=name)]
+
+    def record(self, *args, **kwargs):
+        args = (self,) + args
+        return Variable.record(*args, **kwargs)
+
+
+class Variable(models.Model):
+    TYPES = (
+        ('Integer', int),
+        ('Floating Point', float),
+        ('Text', unicode),
+        ('Flag', bool),
+    )
+    TYPE_CHOICES = (
+        (i, label)
+        for (i, label) in
+        enumerate((label for (label, coercer) in TYPES), 1)
+    )
+
+    session = models.ForeignKey(Session)
+    timestamp = models.DateTimeField(auto_now_add=True)
+    name = models.CharField(max_length=32)
+    type = models.IntegerField(choices=TYPES)
+    value = models.CharField(max_length=130)
+
+    def get_value(self):
+        try:
+            f = float(self.value)
+        except ValueError:
+            f = None
+        try:
+            i = int(self.value)
+        except ValueError:
+            i = None
+        if i is not None and i == f:
+            return i
+        elif f is not None:
+            return f
+        elif self.value == 'true':
+            return True
+        elif self.value == 'false':
+            return False
+        return self.value
```

Comment:
@ironfroggy I am wondering whether type can be used here to return value from the encoded string rather than using type coercion here.